sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -452,22 +455,173 @@ def fetch_all_values(cls, cursor, counters_list, logger):
 
     def fetch_metric(self, rows, columns):
         database_name = columns.index("name")
+        db_state_desc_index = columns.index("state_desc")
         value_column_index = columns.index(self.column)
 
         for row in rows:
             if row[database_name] != self.instance:
                 continue
 
             column_val = row[value_column_index]
-
+            db_state_desc = row[db_state_desc_index]
             metric_tags = [
                 'database:{}'.format(str(self.instance)),
+                'database_state_desc:{}'.format(str(db_state_desc)),
             ]
             metric_tags.extend(self.tags)
             metric_name = '{}'.format(self.datadog_name)
             self.report_function(metric_name, column_val, tags=metric_tags)
 
 
+# https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-hadr-database-replica-states-transact-sql?view=sql-server-ver15
+class SqlDbReplicaStates(BaseSqlServerMetric):
+    TABLE = 'sys.dm_hadr_database_replica_states'
+    DEFAULT_METRIC_TYPE = 'gauge'
+    QUERY_BASE = "select *  \
+                 from \
+                 {table} as dhdrs  \
+                 inner join sys.availability_groups as ag on  \
+                 ag.group_id = dhdrs.group_id \
+                 inner join sys.availability_replicas as ar \
+                 on dhdrs.replica_id = ar.replica_id".format(
+        table=TABLE
+    )
```

Comment:
A lot of the `QUERY_BASE` for the new AlwaysOn metrics ended up looking pretty similar since I needed to join many of the same tables together. Maybe I should just have a single class instead of a few classes to optimize?